chacra/metrics.py

```diff
@@ -58,11 +58,17 @@ def my_oddly_named_function():
 
 def short_hostname(_socket=None):
     """
-    Obtains remote hostname of the socket and cuts off the domain part
+    Returns the config option ``short_hostname`` if found.
+
+    If ``short_hostname`` is not defined, it obtains the
+    remote hostname of the socket and cuts off the domain part
     of its FQDN.
     """
-    _socket = _socket or socket
-    return _socket.gethostname().split('.', 1)[0]
+    short_hostname = getattr(pecan.conf, 'short_hostname')
```

Comment:
this will break with an `AttributeError` because the fallback is not being specified. I think that what you wanted here was:

    short_hostname = getattr(pecan.conf, 'short_hostname', None)

If tests passed with the current implementation I think one is needed where this setting is not available